src/pyqasm/decomposer.py

```diff
@@ -159,3 +170,70 @@ def _get_qubits_for_gate(cls, qubits, rule):
             else:
                 qubits = [qubits[1]]
         return qubits
+
+    @classmethod
+    def _process_rotational_gate(cls, gate_name, statement, target_basis_set, depth, accuracy):
+        """Process the rotational gates based on the target basis set.
+        
+        Args:
+            gate_name: The name of the gate.
+            statement: The statement to process.
+            target_basis_set: The target basis set to rebase the module to.
+            depth: The depth of the approximation.
+            accuracy: The accuracy of the approximation.
+            
+        Returns:
+            list: The processed gates based on the target basis set.
+        """
+        theta = statement.arguments[0].value
+        processed_gates_list = []
+
+        # Use lookup table if ∅ is pi, pi/2 or pi/4
+        if theta in [CONSTANTS_MAP["pi"], CONSTANTS_MAP["pi"] / 2, CONSTANTS_MAP["pi"] / 4]:
+            rotational_lookup_rules = ROTATIONAL_LOOKUP_RULES[target_basis_set]
+            rule_list = rotational_lookup_rules[gate_name][theta]
+            processed_gates_list = cls._get_decomposed_gates(rule_list, statement)
+
+        # Use Solovay-Kitaev's Algorithm for gate approximation
+        else:
+            target_matrix = cls._get_target_matrix_for_rotational_gates(gate_name, theta)
+            approximated_gates = solovay_kitaev(target_matrix, target_basis_set, depth, accuracy)
+
+            for approximated_gate_name in approximated_gates.name:
+                new_gate = qasm3_ast.QuantumGate(
+                    modifiers=[],
+                    name=qasm3_ast.Identifier(name=approximated_gate_name),
+                    arguments=[],
+                    qubits=statement.qubits,
+                )
+
+                processed_gates_list.append(new_gate)
+        return processed_gates_list
+
+    @classmethod
+    def _get_target_matrix_for_rotational_gates(cls, gate_name, theta):
```

Comment:
I believe this method / mapping should be present in the `pyqasm/maps/gates.py` file as it is unrelated to the `Decomposer` logic.